src/inc_gauged_raz.rs

```diff
@@ -1151,4 +1152,30 @@ mod tests {
 
 		assert_eq!(original_names, new_names);
 	}
+
+
+	#[test]
+	fn test_name_indexes() {
+		use raz_meta::Names;
+		use rand::{thread_rng,Rng};
+
+		let mut r: Raz<_,Names> = Raz::new();
+		for i in 0..1000 {
+			r.push_left(i);
+			if i % 10 == 0 {
+				r.archive_left(::inc_level(),Some(name_of_usize(i)));
+			}	
+		}
+		let t = r.unfocus();
+
+		println!("top: {:?}", t.meta());
+
+		for _ in 0..10 {
+			let val = (thread_rng().gen::<usize>() % 100) * 10;
+			let tree_name = name_pair(name_of_usize(val),name_of_string(String::from("tree")));
```

Comment:
OK, sounds good.

BTW: Over this weekend, I started integrating dynamic type checks using `TypeId` into the Adapton engine.  Some, but not all, of the possible dynamic type errors are now checked, and will give a meaningful `panic!` message.  (It will detect the type error, and report the offending Art's location in the message).